torch/csrc/distributed/c10d/comm.h

```diff
@@ -13,4 +14,32 @@ void broadcast_coalesced(
     at::TensorList tensors,
     size_t buffer_size);
 
+class GradBucket {
```

Comment:
Not entirely sure about this, but could it be related to a possible extension on parameter to bucket mappings? From the DDP communication issue, I recall:
```
class GradBucket:
  Tensor bucket_data
  // Optionally in the future this can be enhanced with parameter to bucket 
  // mappings as well.
```
Any thoughts @mrshenli and @zhaojuanmao ?

Reply:
@pritamdamania87 any comments?

Reply:
I think the internal `Bucket` structure for Reducer should be different from what we expose to users here. Decoupling the two allows us to expose different level of information in both places and customize each as needed.